### PR TITLE
fix checksum on siva files and make consistent checksums

### LINKED ISSUE #671: Crash on start with siva files
```
 pga list -l python -f json | head -n 1 | jq -r '.sivaFilenames[]' | pga get -i -o repos

./gitbase server -v -d /tmp/repos/ -i /tmp/index

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x184cab4]

goroutine 1 [running]:
github.com/src-d/gitbase.readRefs(0x2035380, 0xc000a27410, 0x14, 0x20, 0x14, 0x0, 0x0)
	/github.com/src-d/gitbase/checksum.go:98 +0x204
github.com/src-d/gitbase.(*checksumable).Checksum(0xc000a27500, 0x0, 0x0, 0x0, 0x0)
	/github.com/src-d/gitbase/checksum.go:32 +0x1ce
github.com/src-d/gitbase/vendor/gopkg.in/src-d/go-mysql-server.v0/sql.(*IndexRegistry).LoadIndexes(0xc000a19ae0, 0xc0004a7aa0, 0x2, 0x2, 0x0, 0x0)
	/github.com/src-d/gitbase/vendor/gopkg.in/src-d/go-mysql-server.v0/sql/index.go:233 +0x7c7
github.com/src-d/gitbase/vendor/gopkg.in/src-d/go-mysql-server%2ev0.(*Engine).Init(0xc0004a7520, 0x1, 0x1)
	/github.com/src-d/gitbase/vendor/gopkg.in/src-d/go-mysql-server.v0/engine.go:121 +0x43
github.com/src-d/gitbase/cmd/gitbase/command.(*Server).buildDatabase(0xc0001382a0, 0xc0004a74a0, 0xc00000f3c0)
	/github.com/src-d/gitbase/cmd/gitbase/command/server.go:225 +0x319
github.com/src-d/gitbase/cmd/gitbase/command.(*Server).Execute(0xc0001382a0, 0xc00026c420, 0x0, 0x6, 0x0, 0x0)
	/github.com/src-d/gitbase/cmd/gitbase/command/server.go:131 +0x23d
github.com/src-d/gitbase/vendor/github.com/jessevdk/go-flags.(*Parser).ParseArgs(0xc00026c3c0, 0xc00003a080, 0x6, 0x6, 0xc000a248b0, 0x1, 0x2, 0xc00012c630, 0xc000480e00)
	/github.com/src-d/gitbase/vendor/github.com/jessevdk/go-flags/parser.go:316 +0x80b
github.com/src-d/gitbase/vendor/github.com/jessevdk/go-flags.(*Parser).Parse(0xc00026c3c0, 0x1d7f997, 0x7, 0x1e4787d, 0x1c, 0x1e4787d)
	/github.com/src-d/gitbase/vendor/github.com/jessevdk/go-flags/parser.go:186 +0x71
main.main()
	/github.com/src-d/gitbase/cmd/gitbase/main.go:52 +0x39d
```

### PR DESCRIPTION
**Pull request by erizocosmico**
Fixes #671

### CODE REVIEW

**Comment by erizocosmico** [comment]
Tests are failing, looking into it

File: checksum.go [hunk 1]
```diff
@@ -79,6 +81,21 @@ func readChecksum(r repository) ([]byte, error) {
...
+}
+
+type references []reference
+
+type byName []reference
+
+func (b byName) Len() int      { return len(b) }
+func (b byName) Swap(i, j int) { b[i], b[j] = b[j], b[i] }
+func (b byName) Less(i, j int) bool {
+	return strings.Compare(b[i].name, b[j].name) < 0
```
Comment (kuba--) [inline checksum.go]: maybe sorting by hash will be more reliable

**Comment by erizocosmico** [comment]
Can't reproduce why it's failing :/ it's ok on my computer, but failing on travis (only on one of the versions, anyway)

**Comment by kuba--** [comment]
on my machine `TestChecksumSiva` doesn't work - debugging

**Comment by erizocosmico** [comment]
@kuba-- don't worry, think I fixed it

**Comment by erizocosmico** [comment]
Fixed

**Pull request merged**